.cbmc-batch/source/make_common_data_structures.c

```diff
@@ -33,6 +33,18 @@ void ensure_byte_buf_has_allocated_buffer_member(struct aws_byte_buf *const buf)
     buf->buffer = bounded_malloc(sizeof(*(buf->buffer)) * buf->capacity);
 }
 
+void ensure_ring_buffer_has_allocated_members(struct aws_ring_buffer *const ring_buf, size_t size) {
+    ring_buf->allocator = can_fail_allocator();
+    ring_buf->allocation = bounded_malloc(sizeof(*(ring_buf->allocation)) * size);
+    size_t position_head;
+    size_t position_tail;
+    __CPROVER_assume(position_head < size);
+    __CPROVER_assume(position_tail < size);
```

Comment:
Also `__CPROVER_assume(position_head < position_tail);`? Although I guess it's great that this works even when that's not assumed...

Reply:
I think that as this is a ring buffer, head and tail wrap around. Therefore we cannot assume anything about their relative positions.